chesster.js

```diff
@@ -617,446 +256,128 @@ controller.hears([
-                callback("failed to get player: " + team_member.name);
-            }
-        });
-    };
-}
 
-controller.hears([
-	'team members'
-],[
-	'direct_mention', 
-	'direct_message'
-], function(bot, message) {
-    bot_exception_handler(bot, message, function(){
-        var self = this;
-        self.team_name = message.text.split(" ").slice(2).join(" ");
-        if(self.team_name && self.team_name != ""){
-	    loadSheet(self, function(){
-	        getMembers(self, function(){
-	            if(self.members.length == 6){
-                        async.series([
-                            playerRatingAsyncJob(self.members[0]),
-                            playerRatingAsyncJob(self.members[1]),
-                            playerRatingAsyncJob(self.members[2]),
-                            playerRatingAsyncJob(self.members[3]),
-                            playerRatingAsyncJob(self.members[4]),
-                            playerRatingAsyncJob(self.members[5]),
-                        ], function(err){
-                            if(!err){
-                                bot.reply(message, prepareMembersResponse(self));
-                            }else{
-                                bot.reply(message, err);
-                            }
-                        });
-                    }else{
-                        bot.reply(message, "I could not find that team.");
-                    }
-                });
+/* team members */
+chesster.hears({
+    middleware: [slack.requiresLeague],
+	patterns: ['team members'],
+    messageTypes: [
+        'direct_mention', 
+        'direct_message'
+    ]
+},
+function(bot, message) {
+    return Q.fcall(function() {
+        var teamName = message.text.split(" ").slice(2).join(" ");
+        if(teamName && teamName != ""){
+            message.league.formatTeamMembersResponse(teamName).then(function(response) {
+                bot.reply(message, response);
             });
         }else{
             bot.reply(message, "Which team did you say? [ team members <team-name> ]. Please try again.");
         }
     });
 });
 
-/* LICHESS STUFF */
-
-function getPlayerByName(name, callback){
-    var url = "http://en.lichess.org/api/user/" + name;
-    fetch_url_into_json(url, callback);
-}
-
-function getClassicalRating(opp, callback){
-  callback(opp.perfs.classical.rating);
-}
-
 /* welcome */
 
-controller.on('user_channel_join', function(bot, message) {
+chesster.on({event: 'user_channel_join'},
+function(bot, message) {
     bot_exception_handler(bot, message, function(){
         if(message.channel == channels.getId("general")){
             bot.reply(message, "Everyone, please welcome the newest member of the " 
                              + "Lichess 45+45 League, <@" + message.user + ">!");
             
             bot.startPrivateConversation(message, function(err, convo){
+                // TODO: the config links references below are hard coded
```

Comment:
When I wrote this, I wasn't sure. I was, however, trying to ensure that no league-specific stuff was hard coded in chesster.js and this is 45+45 league specific.  

Since then I've thought about this and realized that this probably needs to change. Lone Wolf signups are being confused by this message when they join when the team tournament is not active.  I would be in favour of making this much shorter and much more about the very basics:

You're new here, We're happy you joined us. We have two leagues and I'm not sure which one you signed up for, but I'd like to tell you about the league you joined. If you want an introduction with the 45+45 team tournament, say: intro team if you want an introduction to the lone wolf tournament, say: intro lonewolf.

And then there would be an 'intro' command that was league specific. And the number of leagues that have intros could be determined at runtime.